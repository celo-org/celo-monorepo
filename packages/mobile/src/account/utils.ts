import { ADDRESS_LENGTH } from 'src/exchange/reducer'

export const isAddressFormat = (content: string): boolean => {
  return content.startsWith('0x') && content.length === ADDRESS_LENGTH
}
