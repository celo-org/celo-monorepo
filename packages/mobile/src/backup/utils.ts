import { sampleSize } from 'lodash'
import { AsyncStorage } from 'react-native'
import { generateMnemonic, wordlists } from 'react-native-bip39'
import { getKey, setKey } from 'src/utils/keyStore'
import Logger from 'src/utils/Logger'

const TAG = 'Backup/utils'

export async function createQuizWordList(mnemonic: string, language: string | null) {
  const disallowedWordSet = new Set(mnemonic.split(' '))
  const languageWordList = getWordlist(language)
  const wordOptions: string = await generateMnemonic(1000, null, languageWordList)
  const quizWordList = new Set(
    [...wordOptions.split(' ')].filter((word: string) => !disallowedWordSet.has(word))
  )
  return [...quizWordList]
}

export function selectQuizWordOptions(correctWord: string, allWords: string[], numOptions: number) {
  const wordOptions = []
  const correctWordPosition = Math.floor(Math.random() * numOptions)
  const randomWordIndexList = sampleSize([...Array(allWords.length).keys()], numOptions - 1)
  let randomWordIndex: number = 0

  for (let i = 0; i < numOptions; i++) {
    if (i === correctWordPosition) {
      wordOptions.push(correctWord)
      continue
    }

    wordOptions.push(allWords[randomWordIndexList[randomWordIndex]])
    randomWordIndex += 1
  }
  return wordOptions
}

export function getWordlist(language: string | null) {
  if (!language) {
    return wordlists.EN
  }

  switch (language.slice(0, 2)) {
    case 'es': {
      return wordlists.ES
    }
    default: {
      return wordlists.EN
    }
  }
}

// TODO(Rossy) Remove after the next alfa testnet reset
export async function getStoredMnemonic(): Promise<string | null> {
  try {
    Logger.debug(TAG, 'Checking keystore for mnemonic')
    let mnemonic = await getKey('mnemonic')
    if (mnemonic) {
      return mnemonic
    }

    Logger.debug(TAG, 'Mnemonic not found in keystore, checking async storage')
    mnemonic = await AsyncStorage.getItem('mnemonic')
    if (mnemonic) {
      await setKey('mnemonic', mnemonic)
      await AsyncStorage.removeItem('mnemonic')
      return mnemonic
    }

    Logger.error(TAG, 'No mnemonic found')
    return null
  } catch (error) {
    Logger.error(TAG, 'Failed to retrieve mnemonic', error)
    return null
  }
}
