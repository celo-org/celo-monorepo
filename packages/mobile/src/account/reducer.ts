import { isE164Number } from '@celo/utils/src/phoneNumbers'
import { Actions, ActionTypes } from 'src/account/actions'
import { PaymentRequest } from 'src/account/types'
import { DEV_SETTINGS_ACTIVE_INITIALLY } from 'src/config'
import { RootState } from 'src/redux/reducers'
import { getRemoteTime } from 'src/utils/time'

export interface State {
  name: string
  e164PhoneNumber: string
  defaultCountryCode: string
  contactDetails: UserContactDetails
  devModeActive: boolean
  devModeClickCount: number
  photosNUXClicked: boolean
  pincodeType: PincodeType
  isSettingPin: boolean
  accountCreationTime: number
  backupCompleted: boolean
  backupDelayedTime: number
  socialBackupCompleted: boolean
  incomingPaymentRequests: PaymentRequest[]
  outgoingPaymentRequests: PaymentRequest[]
  dismissedEarnRewards: boolean
  dismissedInviteFriends: boolean
}

export enum PincodeType {
  Unset = 'Unset',
  PhoneAuth = 'PhoneAuth',
  CustomPin = 'CustomPin',
}

export interface UserContactDetails {
  contactId: string | null
  thumbnailPath: string | null
}

export const initialState = {
  name: '',
  e164PhoneNumber: '',
  defaultCountryCode: '',
  contactDetails: {
    contactId: null,
    thumbnailPath: null,
  },
  devModeActive: DEV_SETTINGS_ACTIVE_INITIALLY,
  devModeClickCount: 0,
  photosNUXClicked: false,
  pincodeType: PincodeType.Unset,
  isSettingPin: false,
  accountCreationTime: 99999999999999,
  incomingPaymentRequests: [],
  outgoingPaymentRequests: [],
  backupCompleted: false,
  backupDelayedTime: 0,
  socialBackupCompleted: false,
  dismissedEarnRewards: false,
  dismissedInviteFriends: false,
}

export const reducer = (state: State | undefined = initialState, action: ActionTypes): State => {
  switch (action.type) {
    case Actions.SET_NAME:
      return {
        ...state,
        name: action.name,
      }
    case Actions.SET_PHONE_NUMBER:
      if (!isE164Number(action.e164PhoneNumber)) {
        return state
      }
      return {
        ...state,
        e164PhoneNumber: action.e164PhoneNumber,
        defaultCountryCode: action.countryCode,
      }
    case Actions.DEV_MODE_TRIGGER_CLICKED:
      const newClickCount = (state.devModeClickCount + 1) % 6
      return {
        ...state,
        devModeClickCount: newClickCount,
        devModeActive: newClickCount >= 3,
      }
    case Actions.PHOTOSNUX_CLICKED:
      return {
        ...state,
        photosNUXClicked: true,
      }
    case Actions.SET_PINCODE:
      return {
        ...state,
        isSettingPin: true,
      }
    case Actions.SET_PINCODE_SUCCESS:
      return {
        ...state,
        pincodeType: action.pincodeType,
        isSettingPin: false,
      }
    case Actions.SET_PINCODE_FAILURE:
      return {
        ...state,
        pincodeType: PincodeType.Unset,
        isSettingPin: false,
      }
    case Actions.SET_ACCOUNT_CREATION_TIME_ACTION:
      return {
        ...state,
        accountCreationTime: getRemoteTime(),
      }
    case Actions.SET_BACKUP_COMPLETED_ACTION:
      return {
        ...state,
        backupCompleted: true,
      }
    case Actions.SET_BACKUP_DELAYED_ACTION:
      return {
        ...state,
        backupDelayedTime: getRemoteTime(),
      }
    case Actions.SET_SOCIAL_BACKUP_COMPLETED_ACTION:
      return {
        ...state,
        socialBackupCompleted: true,
      }
    case Actions.RESET_BACKUP_STATE:
      return {
        ...state,
        backupCompleted: false,
        socialBackupCompleted: false,
        backupDelayedTime: 0,
      }
    case Actions.UPDATE_INCOMING_PAYMENT_REQUESTS:
      return {
        ...state,
        incomingPaymentRequests: action.paymentRequests,
      }
    case Actions.UPDATE_OUTGOING_PAYMENT_REQUESTS:
      return {
        ...state,
        outgoingPaymentRequests: action.paymentRequests,
      }
    case Actions.DISMISS_EARN_REWARDS:
      return {
        ...state,
        dismissedEarnRewards: true,
      }
    case Actions.DISMISS_INVITE_FRIENDS:
      return {
        ...state,
        dismissedInviteFriends: true,
      }
    case Actions.SET_USER_CONTACT_DETAILS:
      return {
        ...state,
        contactDetails: {
          contactId: action.contactId,
          thumbnailPath: action.thumbnailPath,
        },
      }
    default:
      return state
  }
}

export const devModeSelector = (state: RootState) => state.account.devModeActive
export const nameSelector = (state: RootState) => state.account.name
export const e164NumberSelector = (state: RootState) => state.account.e164PhoneNumber
export const defaultCountryCodeSelector = (state: RootState) => state.account.defaultCountryCode
export const userContactDetailsSelector = (state: RootState) => state.account.contactDetails
export const pincodeTypeSelector = (state: RootState) => state.account.pincodeType
