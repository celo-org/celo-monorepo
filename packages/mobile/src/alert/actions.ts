import { TOptions } from 'i18next'
import { ErrorDisplayType } from 'src/alert/reducer'
import { AnalyticsEvents } from 'src/analytics/Events'
import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
import { ErrorMessages } from 'src/app/ErrorMessages'
import { ALERT_BANNER_DURATION } from 'src/config'
import i18n, { Namespaces } from 'src/i18n'

export enum Actions {
  SHOW = 'ALERT/SHOW',
  HIDE = 'ALERT/HIDE',
}

enum AlertTypes {
  MESSAGE = 'message',
  ERROR = 'error',
}

interface ShowAlertAction {
  type: Actions.SHOW
  alertType: AlertTypes
  displayMethod: ErrorDisplayType
  message: string
  dismissAfter?: number | null
  buttonMessage?: string | null
  title?: string | null
  underlyingError?: ErrorMessages | null
}

export const showMessage = (
  message: string,
  dismissAfter?: number | null,
  buttonMessage?: string | null,
  title?: string | null
): ShowAlertAction => {
  return showAlert(AlertTypes.MESSAGE, message, dismissAfter, buttonMessage, title)
}

export const showError = (
  error: ErrorMessages,
  dismissAfter?: number | null,
  i18nOptions?: object
): ShowAlertAction => {
  ValoraAnalytics.track(AnalyticsEvents.error_displayed, { error })
  return showAlert(
    AlertTypes.ERROR,
    i18n.t(error, { ns: 'global', ...(i18nOptions || {}) }),
    dismissAfter,
    null,
    null,
    error
  )
}

export const showErrorInline = (error: ErrorMessages, i18nOptions?: TOptions): ShowAlertAction => ({
  type: Actions.SHOW,
  alertType: AlertTypes.ERROR,
  displayMethod: ErrorDisplayType.INLINE,
  message: i18n.t(error, { ns: Namespaces.global, ...(i18nOptions || {}) }),
  underlyingError: error,
})

// Useful for showing a more specific error if its a documented one, with
// a fallback to something more generic
export function showErrorOrFallback(error: any, fallback: ErrorMessages) {
  if (error && Object.values(ErrorMessages).includes(error.message)) {
    return showError(error.message)
  }

  ValoraAnalytics.track(AnalyticsEvents.error_fallback, {
    error: fallback,
  })

  return showError(fallback)
}

const showAlert = (
  alertType: AlertTypes,
  message: string,
  dismissAfter: number | null = ALERT_BANNER_DURATION,
  buttonMessage?: string | null,
  title?: string | null,
  underlyingError?: ErrorMessages | null
): ShowAlertAction => {
  return {
    type: Actions.SHOW,
    alertType,
    displayMethod: ErrorDisplayType.BANNER,
    message,
    dismissAfter,
    buttonMessage,
    title,
    underlyingError,
  }
}

interface HideAlertAction {
  type: Actions.HIDE
}

export const hideAlert = (): HideAlertAction => ({
  type: Actions.HIDE,
})

export type ActionTypes = ShowAlertAction | HideAlertAction
