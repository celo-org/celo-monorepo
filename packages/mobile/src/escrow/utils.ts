import { ensureLeading0x } from '@celo/base'
import { generateDeterministicInviteCode } from '@celo/utils/lib/account'
import { publicKeyToAddress } from '@celo/utils/lib/address'

export const generateEscrowPaymentIdAndPk = (
  recipientPhoneHash: string,
  recipientPepper: string,
  addressIndex: number = 0
) => {
  const { publicKey, privateKey } = generateDeterministicInviteCode(
    recipientPhoneHash,
    recipientPepper,
    addressIndex
  )
  return { paymentId: publicKeyToAddress(publicKey), privateKey: ensureLeading0x(privateKey) }
}

export const generateUniquePaymentId = (
  existingPaymentIds: string[],
  phoneHash: string,
  pepper: string
) => {
  const paymentIdSet: Set<string> = new Set(existingPaymentIds)

  // Using an upper bound of 100 to be sure this doesn't run forever given
  // the realistic amount of pending escrow txs is far less than this
  for (let i = 0; i < 100; i += 1) {
    const { paymentId } = generateEscrowPaymentIdAndPk(phoneHash, pepper, i)
    if (!paymentIdSet.has(paymentId)) {
      return paymentId
    }
  }
}
