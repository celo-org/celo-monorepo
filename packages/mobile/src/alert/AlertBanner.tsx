import SmartTopAlert, { NotificationTypes } from '@celo/react-components/components/SmartTopAlert'
import * as React from 'react'
import { connect } from 'react-redux'
import { hideAlert } from 'src/alert/actions'
import { State as AlertState } from 'src/alert/reducer'
import { RootState } from 'src/redux/reducers'

interface StateProps {
  alert: AlertState
}

interface DispatchProps {
  hideAlert: typeof hideAlert
}

type Props = StateProps & DispatchProps

const mapStateToProps = (state: RootState): StateProps => {
  return {
    alert: state.alert,
  }
}

const mapDispatchToProps = {
  hideAlert,
}

export class AlertBanner extends React.Component<Props> {
  render() {
    const { alert, hideAlert: hideAlertAction } = this.props

    return (
      <SmartTopAlert
        timestamp={Date.now()}
        text={alert && alert.message}
        onPress={hideAlertAction}
        type={alert && alert.type === 'error' ? NotificationTypes.ERROR : NotificationTypes.MESSAGE}
        dismissAfter={alert && alert.dismissAfter}
        buttonMessage={alert && alert.buttonMessage}
        title={alert && alert.title}
      />
    )
  }
}

export default connect<StateProps, DispatchProps, {}, RootState>(
  mapStateToProps,
  mapDispatchToProps
)(AlertBanner)
