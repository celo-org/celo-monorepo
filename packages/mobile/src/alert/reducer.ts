import { Actions, ActionTypes, AlertTypes, ShowAlertAction } from 'src/alert/actions'
import { ErrorMessages } from 'src/app/ErrorMessages'
import { ALERT_BANNER_DURATION } from 'src/config'
import {
  Actions as ExchangeActions,
  ActionTypes as ExchangeActionTypes,
} from 'src/exchange/actions'
import i18n from 'src/i18n'
import { RootState } from 'src/redux/reducers'

export enum ErrorDisplayType {
  'BANNER',
  'INLINE',
}

export type State = {
  type: 'message' | 'error'
  displayMethod: ErrorDisplayType
  message: string
  dismissAfter?: number | null
  buttonMessage?: string | null
  action?: object | null
  title?: string | null
  underlyingError?: ErrorMessages | null
} | null

const initialState = null

const errorAction = (error: ErrorMessages): ShowAlertAction => ({
  type: Actions.SHOW,
  alertType: AlertTypes.ERROR,
  displayMethod: ErrorDisplayType.BANNER,
  message: i18n.t(error, { ns: 'global' }),
  dismissAfter: ALERT_BANNER_DURATION,
  buttonMessage: null,
  title: null,
  underlyingError: error,
})

export const reducer = (
  state: State = initialState,
  action: ActionTypes | ExchangeActionTypes
): State => {
  switch (action.type) {
    case ExchangeActions.WITHDRAW_CELO_FAILED:
      action = errorAction(action.error)
    case Actions.SHOW:
      return {
        displayMethod: action.displayMethod,
        type: action.alertType,
        message: action.message,
        dismissAfter: action.dismissAfter,
        buttonMessage: action.buttonMessage,
        action: action.action,
        title: action.title,
        underlyingError: action.underlyingError,
      }
    case Actions.HIDE:
      return null
    default:
      if (state?.action === action) {
        // Hide alert when the alert action is dispatched
        return null
      }
      return state
  }
}

export const errorSelector = (state: RootState) => {
  return state.alert ? state.alert.underlyingError || null : null
}
