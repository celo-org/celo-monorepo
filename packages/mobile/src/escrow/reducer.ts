import { createSelector } from 'reselect'
import { Actions, ActionTypes, EscrowedPayment } from 'src/escrow/actions'
import { RootState } from 'src/redux/reducers'

export interface State {
  isReclaiming: boolean
  sentEscrowedPayments: EscrowedPayment[]
}

export const initialState = {
  isReclaiming: false,
  sentEscrowedPayments: [],
}

export const escrowReducer = (state: State | undefined = initialState, action: ActionTypes) => {
  switch (action.type) {
    case Actions.STORE_SENT_PAYMENTS:
      return {
        ...state,
        sentEscrowedPayments: action.sentPayments,
      }
    case Actions.RECLAIM_PAYMENT:
      return {
        ...state,
        isReclaiming: true,
      }
    case Actions.RECLAIM_PAYMENT_FAILURE:
    case Actions.RECLAIM_PAYMENT_SUCCESS:
      return {
        ...state,
        isReclaiming: false,
      }
    default:
      return state
  }
}

export const getSentEscrowPayments = (state: RootState) => state.escrow.sentEscrowedPayments

export const getReclaimableEscrowPayments = createSelector(
  getSentEscrowPayments,
  (sentPayments) => {
    const currUnixTime = Date.now() / 1000
    return sentPayments.filter((payment) => {
      const paymentExpiryTime = +payment.timestamp + +payment.expirySeconds
      return currUnixTime >= paymentExpiryTime
    })
  }
)
