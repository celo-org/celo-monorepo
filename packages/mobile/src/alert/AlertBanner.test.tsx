import * as React from 'react'
import { fireEvent, render } from 'react-native-testing-library'
import { Provider } from 'react-redux'
import AlertBanner from 'src/alert/AlertBanner'
import { ErrorDisplayType } from 'src/alert/reducer'
import { createMockStore } from 'test/utils'

describe('AlertBanner', () => {
  describe('when message passed in', () => {
    it('renders message', () => {
      const { toJSON } = render(
        <Provider
          store={createMockStore({
            alert: {
              type: 'message',
              displayMethod: ErrorDisplayType.BANNER,
              message: 'This is your shadow speaking',
              dismissAfter: 0,
            },
          })}
        >
          <AlertBanner />
        </Provider>
      )
      expect(toJSON()).toMatchSnapshot()
    })
  })

  describe('when message and title passed in', () => {
    it('renders title with message', () => {
      const { toJSON } = render(
        <Provider
          store={createMockStore({
            alert: {
              type: 'message',
              displayMethod: ErrorDisplayType.BANNER,
              title: 'Declaration',
              message: 'This is your shadow speaking',
              dismissAfter: 0,
            },
          })}
        >
          <AlertBanner />
        </Provider>
      )
      expect(toJSON()).toMatchSnapshot()
    })
  })

  describe('when error message passed in', () => {
    it('renders error message', () => {
      const { toJSON } = render(
        <Provider
          store={createMockStore({
            alert: {
              type: 'error',
              displayMethod: ErrorDisplayType.BANNER,
              message: 'This is an error',
              dismissAfter: 0,
            },
          })}
        >
          <AlertBanner />
        </Provider>
      )
      expect(toJSON()).toMatchSnapshot()
    })
  })

  describe('when an action is provided', () => {
    it('it dispatches the action when pressed', () => {
      const store = createMockStore({
        alert: {
          type: 'message',
          displayMethod: ErrorDisplayType.BANNER,
          message: 'My message',
          dismissAfter: 0,
          action: { type: 'MY_ACTION' },
        },
      })
      const { toJSON, getByTestId } = render(
        <Provider store={store}>
          <AlertBanner />
        </Provider>
      )
      expect(toJSON()).toMatchSnapshot()

      fireEvent.press(getByTestId('SmartTopAlertTouchable'))
      expect(store.getActions()).toEqual([{ type: 'MY_ACTION' }])
    })
  })
})
