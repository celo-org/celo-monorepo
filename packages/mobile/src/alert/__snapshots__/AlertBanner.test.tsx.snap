// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertBanner when error message passed in renders error message 1`] = `
<View
  style={
    Object {
      "overflow": "hidden",
    }
  }
  testID="errorBanner"
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FD785B",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingBottom": 10,
        "paddingHorizontal": 25,
        "paddingTop": 10,
        "transform": Array [
          Object {
            "translateY": -500,
          },
        ],
      }
    }
  >
    <View
      style={
        Object {
          "marginHorizontal": 5,
        }
      }
    >
      <svg
        fill="none"
        height="16"
        style={Object {}}
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M10.1569 4.75L11.2175 5.81066L8.97834 8.04984L11.3354 10.4069L10.2747 11.4675L7.91768 9.1105L5.56066 11.4675L4.5 10.4069L6.85702 8.04984L4.61785 5.81067L5.67851 4.75001L7.91768 6.98918L10.1569 4.75Z"
          fill="#FFFFFF"
          fillRule="evenodd"
          style={Object {}}
        />
        <circle
          cx="8"
          cy="8"
          r="6.5"
          stroke="white"
          style={Object {}}
        />
      </svg>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#2E3338",
            "fontFamily": "Hind-Regular",
            "fontSize": 14,
            "lineHeight": 18,
          },
          Object {
            "color": "white",
            "lineHeight": undefined,
            "textAlign": "center",
          },
          Object {
            "fontFamily": "Hind-SemiBold",
          },
        ]
      }
    >
      This is an error
    </Text>
  </View>
</View>
`;

exports[`AlertBanner when message and title passed in renders title with message 1`] = `
<View
  style={
    Object {
      "overflow": "hidden",
    }
  }
  testID="infoBanner"
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#3C9BF4",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingBottom": 10,
        "paddingHorizontal": 25,
        "paddingTop": 10,
        "transform": Array [
          Object {
            "translateY": -500,
          },
        ],
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#2E3338",
            "fontFamily": "Hind-Regular",
            "fontSize": 14,
            "lineHeight": 18,
          },
          Object {
            "color": "white",
            "lineHeight": undefined,
            "textAlign": "center",
          },
          false,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "lineHeight": undefined,
              "textAlign": "center",
            },
            Object {
              "fontFamily": "Hind-SemiBold",
            },
          ]
        }
      >
         
        Declaration
         
      </Text>
      This is your shadow speaking
    </Text>
  </View>
</View>
`;

exports[`AlertBanner when message passed in renders message 1`] = `
<View
  style={
    Object {
      "overflow": "hidden",
    }
  }
  testID="infoBanner"
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#3C9BF4",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingBottom": 10,
        "paddingHorizontal": 25,
        "paddingTop": 10,
        "transform": Array [
          Object {
            "translateY": -500,
          },
        ],
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#2E3338",
            "fontFamily": "Hind-Regular",
            "fontSize": 14,
            "lineHeight": 18,
          },
          Object {
            "color": "white",
            "lineHeight": undefined,
            "textAlign": "center",
          },
          false,
        ]
      }
    >
      This is your shadow speaking
    </Text>
  </View>
</View>
`;
